momento-sdk/src/main/java/momento/sdk/AuthClient.java

```diff
@@ -0,0 +1,44 @@
+package momento.sdk;
+
+import java.util.concurrent.CompletableFuture;
+import momento.sdk.auth.accessControl.DisposableTokenScope;
+import momento.sdk.auth.accessControl.ExpiresIn;
+import momento.sdk.exceptions.InvalidArgumentException;
+import momento.sdk.responses.GenerateDisposableTokenResponse;
+
+public class AuthClient implements IAuthClient {
+  private final TokenClient tokenClient;
+
+  public AuthClient(TokenClient tokenClient) {
+    this.tokenClient = tokenClient;
+  }
```

Comment:
the examples now show "best practice" for creating a cache client is to use the new `CacheClient.create` factory function. We did this to support eager connections, which probably isn't as relevant for AuthClient, but I think we probably want the best practice code paths for constructing them to look similar, so we should probably add a `create` method here as well. Any thoughts @nand4011 ?